### PR TITLE
Upgrade to PHPUnit 8.5

### PR DESCRIPTION
**Pull request by emmanix2002**
Description
---
- [x] Updated the custom constraints to use the correct class name
- [x] Removed syntaxCheck attribute because https://stackoverflow.com/a/44331140/665776
- [x] Updated beStrictAboutTestSize see https://github.com/sebastianbergmann/phpunit/issues/3048
- [x] Added more ignore paths

### CODE REVIEW

**Comment by emmanix2002** [comment]
@AhmedRaafat14 I've made the fixes to basically allow PHPUnit not crash on run but I've run into another issue: PHPUnit doesn't execute any tests.

I'm currently investigating why.

**What I've discovered so far**
I've traced it up to the point where I find that the files in the test suite are identified by the FileIterator; currently looking into what happens from [here](https://github.com/sebastianbergmann/phpunit/blob/3f9c4079d1407cd84c51c02c6ad1df6ec2ed1348/src/Util/Configuration.php#L1005)

**Comment by emmanix2002** [comment]
> @AhmedRaafat14 I've made the fixes to basically allow PHPUnit not crash on run but I've run into another issue: PHPUnit doesn't execute any tests.
> 
> I'm currently investigating why.
> 
> **What I've done so far**
> I've traced it up to the point where I find that the files in the test suite are identified by the FileIterator; currently looking into what happens from [here](https://github.com/sebastianbergmann/phpunit/blob/3f9c4079d1407cd84c51c02c6ad1df6ec2ed1348/src/Util/Configuration.php#L1005)

@AhmedRaafat14 this was resolved as soon as I changed PHPUnit versions

File: tests/PHPUnit/PredisTestCase.php [hunk 1]
```diff
@@ -337,4 +337,9 @@ protected function checkRequirements()
 
         $this->checkRequiredRedisServerVersion();
     }
+    
+    public function toString(): string
```
Comment (AhmedRaafat14) [inline tests/PHPUnit/PredisTestCase.php:341]: I believe this should be `__toString` if am not mistaken!

**Comment by AhmedRaafat14** [comment]
@emmanix2002 Cool!

Good job man 👏.

I see you still working on fixing the tests if you want to divide them between me & you just let me know and I will always follow this PR :).

File: tests/PHPUnit/PredisTestCase.php (hunk 1 above)
Comment (emmanix2002) [reply tests/PHPUnit/PredisTestCase.php:341]: The method defined on the interface is actually `toString`

**Comment by srmklive** [comment]
I have all the tests. Except one :smirk:. See #639

**Comment by emmanix2002** [comment]
> I have all the tests. Except one 😏. See #639

@srmklive AWESOME!!!!!!

**Comment by emmanix2002** [comment]
I'll close this PR.

**Pull request closed without merging**